MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpress.c

```diff
@@ -166,6 +167,15 @@ EnumerateNvmeDevNamespace (
     Device->BlockIo2.FlushBlocksEx = NvmeBlockIoFlushBlocksEx;
     InitializeListHead (&Device->AsyncQueue);
 
+    //
+    // Create Media Sanitize Protocol instance
+    //
+    Device->MediaSanitize.Revision   = MEDIA_SANITIZE_PROTOCOL_REVISION;
+    Device->MediaSanitize.Media      = &Device->Media;
+    Device->MediaSanitize.MediaClear = NvmExpressMediaClear;
+    Device->MediaSanitize.MediaPurge = NvmExpressMediaPurge;
+    CopyMem (&(Device->MediaSanitize.SanitizeCapabilities), &(Device->Controller->ControllerData->Sanicap), sizeof (NVME_SANICAP));

```

Comment:
this should be sizeof(Device->MediaSanitize.SanitizeCapabilities)  or sizeof(UINT32)  because you want to make sure you never overflow the destination.  

That said a StaticAssert might also be good as you want to catch the case when they are different and break the build.

Reply:
Agreed... and updated to user sizeof(Device->MediaSanitize.SanitizeCapabilities). Also added an assert in the line before to check that both var sizes are equal.